Adding branch description to Contributing

josenavas opened the pull request:
Adding a description of the branches to the CONTRIBUTING.md file

antgonza commented:
:+1:

ElDeveloper commented on CONTRIBUTING.md:
```diff
@@ -17,6 +17,14 @@ always take into consideration how these new features affect users and whether
 or not adding a new section or document to the documentation (found under the
 `doc` folder) would be useful.
 
+###Repository branch structure
```
I just noticed that the `###` is not being rendered as a title, see [here](https://github.com/josenavas/QiiTa/blob/73dca6fc78ca29a70725667ca916f7464618acd0/CONTRIBUTING.md). Seems to be a problem with a missing space between `###` and the first letter in the title.

ElDeveloper commented on CONTRIBUTING.md:
```diff
@@ -17,6 +17,14 @@ always take into consideration how these new features affect users and whether
+
+* `master`: This branch reflects the code deployed in our [main Qiita server](http://qiita.microbio.me).
+* `dev`: This branch is the active development branch. All new Pull Requests should be performed against this branch.
+* `release-candidate`: This branch is used to freeze the code from the `dev` branch, so we can deploy in our test servers and exercise the code extensively before deploying in our main system.
```
Do we want to make the test server URL public? Also, would it be worth noting the timeline?

ElDeveloper replied:
By timeline I mean how long each branch is alive for, when are freezes happening, etc.

jdereus commented:
I think our test server(s) need to remain internal.  At the least not published.
Should be fire walled to only allow certain access.

As the test server code should only be slightly ahead of production, functionality should be fairly similar in most cases.

From: Yoshiki Vázquez Baeza <notifications@github.com<mailto:notifications@github.com>>
Reply-To: biocore/qiita <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, August 29, 2017 at 8:02 AM
To: biocore/qiita <qiita@noreply.github.com<mailto:qiita@noreply.github.com>>
Cc: Subscribed <subscribed@noreply.github.com<mailto:subscribed@noreply.github.com>>
Subject: Re: [biocore/qiita] Adding branch description to Contributing (#2256)


@ElDeveloper approved this pull request.

________________________________

In CONTRIBUTING.md<https://github.com/biocore/qiita/pull/2256#discussion_r135817204>:

> @@ -17,6 +17,14 @@ always take into consideration how these new features affect users and whether
 or not adding a new section or document to the documentation (found under the
 `doc` folder) would be useful.

+###Repository branch structure


I just noticed that the ### is not being rendered as a title, see here<https://github.com/josenavas/QiiTa/blob/73dca6fc78ca29a70725667ca916f7464618acd0/CONTRIBUTING.md>. Seems to be a problem with a missing space between ### and the first letter in the title.

________________________________

In CONTRIBUTING.md<https://github.com/biocore/qiita/pull/2256#discussion_r135817547>:

> @@ -17,6 +17,14 @@ always take into consideration how these new features affect users and whether
 or not adding a new section or document to the documentation (found under the
 `doc` folder) would be useful.

+###Repository branch structure
+
+The Qiita repository contains three branches:
+
+* `master`: This branch reflects the code deployed in our [main Qiita server](http://qiita.microbio.me).
+* `dev`: This branch is the active development branch. All new Pull Requests should be performed against this branch.
+* `release-candidate`: This branch is used to freeze the code from the `dev` branch, so we can deploy in our test servers and exercise the code extensively before deploying in our main system.


Do we want to make the test server URL public? Also, would it be worth noting the timeline?

-
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/biocore/qiita/pull/2256#pullrequestreview-59272366>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANq27aqmV1h0ngb4d8C1GXaI-TGzkW6Jks5sdCgCgaJpZM4PGEi6>.

josenavas commented:
Thanks @ElDeveloper !

I've fixed the rendering error and I've added a comment about timelines. I agree with @jdereus on keeping the URL of the test systems internal.

ElDeveloper commented:
> I think our test server(s) need to remain internal.  At the least not published.
Should be fire walled to only allow certain access.

Sounds good.

Thanks for making the changes @josenavas
